code/modules/mob/living/living.dm

```diff
@@ -1707,6 +1723,23 @@
 	. = usable_legs
 	usable_legs = new_value
 
+	if(new_value > .) // Gained leg usage.
+		REMOVE_TRAIT(src, TRAIT_FLOORED, LACKING_LOCOMOTION_APPENDAGES_TRAIT)
+		REMOVE_TRAIT(src, TRAIT_IMMOBILIZED, LACKING_LOCOMOTION_APPENDAGES_TRAIT)
+	else if(!(movement_type & (FLYING | FLOATING))) //Lost leg usage, not flying.
+		if(!usable_legs)
+			ADD_TRAIT(src, TRAIT_FLOORED, LACKING_LOCOMOTION_APPENDAGES_TRAIT)
+			if(!usable_hands)
+				ADD_TRAIT(src, TRAIT_IMMOBILIZED, LACKING_LOCOMOTION_APPENDAGES_TRAIT)
+
+	if(usable_legs < default_num_legs)
+		var/limbless_slowdown = (default_num_legs * 3) - (usable_legs * 3)
+		if(!usable_legs && usable_hands < default_num_hands)
+			limbless_slowdown += (default_num_hands * 3) - (usable_hands * 3)
```

Comment:
See previous review of a similar block, the multipliers should be scalars, is there a reason that and this have differences? can you explain it?

Reply:
No particular reason, either I wasn't paying attention or I copypasted code. Fixing it.